docs/DeveloperGuide.md

```diff
@@ -281,47 +296,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+
+**Use case: Getting help with the application**
+
+**MSS**
+1. User needs help in utilizing RC4HDB
+2. User enters the help command
+3. RC4HDB displays a message that directs the user to our user guide
+   
+    Use case ends.
+
+**Use case: Adding a single resident**
+
+**MSS**
+1. New resident moves into RC4
+2. User gets personal details of the resident
+3. User enters the add command and enters the resident information in the required format
+4. RC4HDB adds the resident to the data file
+5. RC4HDB displays the name and other information of the resident
+   
+    Use case ends.
+
+**Extensions**
+
+* 3a. User enters resident information in an invalid format
+  
+    * 3a1. RC4HDB shows an error message.
+      
+      Use case resumes at step 3.
 
-**Use case: Delete a person**
+**Use case: Listing out information of all residents**
 
 **MSS**
+1. User wants to see the full list of residents in RC4
+2. User enters the list command
+3. RC4HDB displays the details of all residents n RC4.
+   
+    Use case ends.
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+**Extension**
+* 2a. The list is empty 
+  
+    Use case ends.
+
+
+**Use case: Editing a single resident’s information**
+
+**MSS**
+1. Resident has a change in personal information
+2. User needs to edit the resident's information in RC4HDB
+3. User looks at which category the information falls under
+4. User enters the edit command and enters the category of the information to be altered, followed by the updated information
+5. RC4HDB updates the information of the specified resident
+6. RC4HDB displays a message detailing the changes made
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 3a. There is no relevant category for that information
+  
+    Use case ends.
+
+
+* 4a. User enters resident information in an invalid format
+
+    * 4a1. RC4HDB shows an error message.
+      
+      Use case resumes at step 3.
+
+
+**Use case: Finding a resident’s information by their name**
+
+**MSS**
+1. User wants to search for a resident’s information
+2. User enters the find command followed by the name of the resident they are looking for
+3. RC4HDB searches the database for the user's input
+4. RC4HDB displays the resident's information
+   
+    Use case ends.
+
+**Extension**
+
+* 3a. RC4HDB cannot find any resident matching the user input
+  
+    * 3a1. RC4HDB shows an error message.
+      
+      Use case resumes at step 2.
+
+
+* 4b. RC4HDB finds multiple residents matching the user input
+    
+    * 4b1. RC4HDB shows a list of all matching residents.
+      
+      Use case ends.
+      
+
+**Use case: Filtering the list of all residents by specific fields**
+
+**MSS**
+1. User want to see a list of residents that fall under a certain category
```

Comment:
Can we change it to "User wants to see..."

Reply:
Thanks for catching that!